pkg/operation/botanist/component/nodeproblemdetector/node_problem_detector.go

```diff
@@ -283,6 +287,11 @@ func (c *nodeProblemDetector) computeResourcesData() (map[string][]byte, error)
 						HostNetwork:                   false,
 						TerminationGracePeriodSeconds: pointer.Int64(daemonSetTerminationGracePeriodSeconds),
 						PriorityClassName:             "system-cluster-critical",
+						SecurityContext: &corev1.PodSecurityContext{
+							SeccompProfile: &corev1.SeccompProfile{
+								Type: corev1.SeccompProfileTypeRuntimeDefault,
+							},
+						},
```

Comment:
Don't we need a handling for kubernetes version here as well?

Reply:
Yes, you are right! I did the handling in one of the next commits